src/nav/ChatNavBar.js

```diff
@@ -54,14 +54,14 @@ export default function ChatNavBar(props: Props) {
           }}
         >
           <NavBarBackButton color={color} />
-          <Title color={color} narrow={narrow} editMessage={editMessage} />
+          <Title color={textColor} narrow={narrow} editMessage={editMessage} />
           <ExtraButton color={color} narrow={narrow} />
           <InfoButton color={color} narrow={narrow} />
         </View>
         <LoadingBanner
-          spinnerColor={spinnerColor}
+          spinnerColor={textColor}
```

Comment:
Huh, yeah; I'd say `LoadingBanner` should always get the same color for its spinner and its text. Probably would be good to change its interface to enforce that, except that we're on the path to reworking/replacing `LoadingBanner` in a way that this detail will become irrelevant; so, needn't bother, I think.

Reply:
It turns out the `LoadingBanner` code is really janky, I think there probably shouldn't be a `default` here at all. But like you say, we're on our way to replacing it, so I'm inclined not to fix it.

Reply:
> It turns out the `LoadingBanner` code is really janky, I think there probably shouldn't be a `default` here at all.

Yeah, possibly we could have done something better with `LoadingBanner` when it was added in #3901.

I don't think I've understood what you've identified as janky, though; should we not have ever used the brand color for the spinner (e.g., as [here](https://github.com/zulip/zulip-mobile/pull/4544#issuecomment-801634074) you've noticed we do), or should that case not have been given "default" semantics (e.g., bfb884a0ec131c9cc02ab664a18d446530c89a5c could have used a name like "brand" instead of "default), or something else? I think `LoadingIndicator`'s `color` prop appropriately takes a very few specific values because it makes a choice between a few specific assets.

Reply:
I can't speak for what Wesley had in mind, but rereading `LoadingBanner` now, one thought I have is that it might be cleaner to have it take just the one parameter `backgroundColor`, and derive the others from that; it has all the information it needs to make the same choices `ChatNavBar` is making. As it stands, it's effectively an implicit part of its interface that those colors need to contrast and coordinate appropriately.

> I think `LoadingIndicator`'s `color` prop appropriately takes a very few specific values because it makes a choice between a few specific assets.

I agree with this reasoning. One further step that might be helpful -- might be helpful even with `LoadingBanner` on its way out, because `LoadingIndicator` has other callers -- is to give a name to that enum `'default' | 'black' | 'white'`. That'd let the same name be used in the types of `SpinningProgress`, `LoadingIndicator`, and `LoadingBanner`, helping make clear the linkage between that prop in their respective interfaces; it'd also provide a place to put a few words of jsdoc about what "default" means here.